st2common/st2common/util/action_output.py

```diff
@@ -0,0 +1,85 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
```

Comment:
I think over time, the team didn't make the st2 software architecture very clean.  I would call them in util modules if these functions are used only by a single st2 component.  If it's a functionality that is required by other st2 components, i kinda prefer them to be part of the service modules.  Having said that, I prefer the main function for validating output to be part of the action service.  In this case, the API and the workflow service would also need to access this.

Reply:
@bigmstone making the point short, can you move these functions to `st2common/services/action.py`?

Reply:
At first, I was thinking that you can implement the action output validation methods in action service and then put the check at https://github.com/StackStorm/st2/blob/master/st2common/st2common/services/action.py#L186. But then looking at what you just pointed me since the validation is going to be called by the action db util, I guess I'm ok with where you put them.  To not block this, just make sure it errors appropriately for orquesta and mistral.